components/raftstore/src/store/peer_storage.rs

```diff
@@ -1594,24 +1543,8 @@ where
         }
 
         self.schedule_applying_snapshot();
-        let prev_region = self.region().clone();
-        self.set_region(snap_region);
-        if self.truncated_index() == self.applied_index() {
-            self.applied_index_term = self.truncated_term();
-        } else {
-            panic!(
-                "{} applied index should be equal to truncated index after snapshot: {} != {}",
-                self.tag,
-                self.applied_index(),
-                self.truncated_index()
-            );
-        }
 
-        Some(ApplySnapResult {
-            prev_region,
-            region: self.region().clone(),
-            destroyed_regions: ctx.destroyed_regions,
-        })
+        self.set_region(res.region.clone());
```

Comment:
This maybe confusing as all other states are updated in place.

Reply:
Indeed, I think updating in place is more similar to the original code.

Reply:
Can you add a comment explaining why delay setting the field?

Reply:
Done.